<launch>

    <!-- Gesture parameters -->
    
    <!-- Speech parameters -->
    <param name="demo" type="int" value="0"/>   <!-- 0: full demo; 3: AWS demo; 7: robot demo -->
    <param name="vad" type="str" value="webrtcvad" />
    <param name="save_speech" type="bool" value="False"/>

    <!-- General parameters -->
    <param name="language" type="str" value="ita"/>  <!-- "ita" or "eng" -->
    <param name="fiware_cb" type="str" value="None"/>  <!-- no publish on FIWARE context broaker -->
    
    <!-- Gesture nodes -->
    <node pkg="gesture_pkg" name="image_input_node" type="image_input_node.py" output="screen"/>
    <node pkg="gesture_pkg" name="hand_gesture_classification_node" type="hand_gesture_classification_node.py" output="screen"/>

    <!-- Speech nodes -->
    <node pkg="speech_pkg" name="classifier" type="speech_classification_node.py" output="screen"/>
    <node pkg="speech_pkg" name="manager" type="manager_node.py" output="screen"/>
    <node pkg="speech_pkg" name="speech_detection_node" type="speech_detection_node.py" output="screen"/>
    <node pkg="speech_pkg" name="microphone_node" type="microphone_node.py" output="screen"/>

    <!-- Webserver node -->
    <node pkg="webview_pkg" name="webview_node" type="webview_node.py" output="screen"/>

</launch>